arduino/cores/board.go

```diff
@@ -139,3 +139,43 @@ func (b *Board) GeneratePropertiesForConfiguration(config string) (*properties.M
 	}
 	return b.GetBuildProperties(fqbn.Configs)
 }
+
+// IsBoardMatchingIDProperties returns true if the board match the given
+// identification properties
+func (b *Board) IsBoardMatchingIDProperties(query *properties.Map) bool {
+	portIDPropsSet := b.Properties.SubTree("upload_port")
+	if portIDPropsSet.Size() == 0 {
+		return false
+	}
+
+	check := func(p *properties.Map) bool {
```

Comment:
It's not exactly clear what this `check` function checks, I'd be more verbose naming it.

Reply:
added a comment, couldn't find a better name :-D